synapse/rest/client/v2_alpha/report_event.py

```diff
@@ -35,16 +39,48 @@ def __init__(self, hs: "HomeServer"):
         super().__init__()
         self.hs = hs
         self.auth = hs.get_auth()
-        self.abuse_report_handler = hs.get_abuse_reporter()
+        self.abuse_report_handler = hs.get_abuse_report_handler()
+        self._rate_limiter = Ratelimiter(
+            store=hs.get_datastore(),
+            clock=hs.get_clock(),
+            rate_hz=1,  # Limit to 1 message per second
+            burst_count=10,  # Limit to 10 successive messages
+        )
 
     async def on_POST(self, request: SynapseRequest, room_id, event_id):
-        logger.debug("on_POST %s" % event_id)
         requester = await self.auth.get_user_by_req(request)
         body = parse_json_object_from_request(request)
-        return await self.abuse_report_handler.report(
-            request, requester.user, body, room_id, event_id
+        await self._rate_limiter.ratelimit(requester=requester)
+
+        if not isinstance(body["reason"], str):
```

Comment:
this will 500 if there is no `reason` property.

```suggestion
        reason = body.get("reason")
        if not isinstance(reason, str):
```

likewise below for `score`.